include/taco/index_notation/index_notation.h

```diff
@@ -1031,6 +1038,18 @@ class IndexVar : public IndexExpr, public IndexVarInterface {
   /// Returns the name of the index variable.
   std::string getName() const;
 
+  size_t& getBound() const;
```

Comment:
Mapping a for all to an index var's bound is not possible since adding schedules replaces the occurrence of the index var from the stmt in certain cases.

Instead, in the updated pull request, we tag this information to a suchThatNode, adding a map that relates the bounded variable, bound size and bound type.